src/Symfony/Component/HttpClient/Retry/MultiplierRetryStrategy.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Retry;
+
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * A retry strategy with a constant or exponential retry delay.
+ *
+ * For example, if $delayMilliseconds=10000 & $multiplier=1 (default),
+ * each retry will wait exactly 10 seconds.
+ *
+ * But if $delayMilliseconds=10000 & $multiplier=2:
+ *      * Retry 1: 10 second delay
+ *      * Retry 2: 20 second delay (10000 * 2 = 20000)
+ *      * Retry 3: 40 second delay (20000 * 2 = 40000)
+ *
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ *
+ * @final
+ */
+class MultiplierRetryStrategy implements RetryStrategyInterface
```

Comment:
This is an implementation of the exponential backoff. I would use that as name rather than `MultiplierRetry`, to reuse the common name of that concept (that people might already know and might look for)

Reply:
Same name as Messenger Component. Technically, when multiplier is `1`, it's a linear retry.

Reply:
well, with `1` is a constant retry. A linear retry might still have a multiplier, except that the computed value for a linear case would be `$multiplier * $retryCount`, not `$multiplier ** $retryCount` :smile: